xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPBaseRenderer.cpp

```diff
@@ -139,32 +146,59 @@ void CRPBaseRenderer::ManageRenderArea(const IRenderBuffer& renderBuffer)
   float screenHeight;
   float screenPixelRatio;
   GetScreenDimensions(screenWidth, screenHeight, screenPixelRatio);
-
-  // Entire target rendering area for the video (including black bars)
-  const CRect viewRect = m_context.GetViewWindow();
-
-  // Calculate pixel ratio and zoom amount
+  CRect viewRect = m_context.GetViewWindow();
+  CRect destRect;
   float pixelRatio = 1.0f;
   float zoomAmount = 1.0f;
-  CRenderUtils::CalculateStretchMode(stretchMode, rotationDegCCW, sourceWidth, sourceHeight,
-                                     screenWidth, screenHeight, pixelRatio, zoomAmount);
-
-  // Calculate destination dimensions
-  CRect destRect;
-  CRenderUtils::CalcNormalRenderRect(viewRect, sourceAspectRatio * pixelRatio, zoomAmount,
-                                     destRect);
-
   m_sourceRect.x1 = 0.0f;
   m_sourceRect.y1 = 0.0f;
   m_sourceRect.x2 = static_cast<float>(sourceWidth);
   m_sourceRect.y2 = static_cast<float>(sourceHeight);
+  if (scaleMode == SCALINGMETHOD::DISPLAY && stretchMode == STRETCHMODE::Original && bhw_filter)
+  {
+    RESOLUTION_INFO m_gameRes=m_context.GetResInfo();
+    m_gameRes.Overscan.left = 0;
+    m_gameRes.Overscan.top = 0;
+    m_gameRes.Overscan.right = sourceWidth;
+    m_gameRes.Overscan.bottom = sourceHeight;
+    m_gameRes.iWidth = sourceWidth;
+    m_gameRes.iHeight = sourceHeight;
+    m_gameRes.iScreenWidth = sourceWidth;
+    m_gameRes.iScreenHeight = sourceHeight;
+    CServiceBroker::GetWinSystem()->SetFullScreen(true,  m_gameRes, false);
```

Comment:
Is it safe to set the system to fullscreen 60 times per second? A function named `ManageRenderArea()` which does just geometry calculations shouldn't call into an external system or have non-render-geometry related side effects.